workflow-templates/im-test-selenium-manual.yml

```diff
@@ -0,0 +1,223 @@
+
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.GITHUB_REF }}
+
+      - name: Set Cache Keys
+        shell: bash
+        run: echo "NUGET_CACHE_KEY=nuget-${{ hashFiles('**/*.csproj') }}" >> $GITHUB_ENV
+                
+      - name: Check for a .NET cache
+        id: has-cache
+        uses: actions/cache@v4
+        with:
+          lookup-only: true
+          key: ${{ env.NUGET_CACHE_KEY }}
+          path: ${{ env.NUGET_CACHE_PATH }}  
+
+      - name: Authenticate with GitHub Packages
+        if: steps.has-cache.outputs.cache-hit != 'true'
+        uses: im-open/authenticate-with-gh-package-registries@v1.1
+        with:
+          github-token: ${{ secrets.READ_PKG_TOKEN }}
+
+      - name: Restore Dependencies
+        if: steps.has-cache.outputs.cache-hit != 'true'
+        shell: bash
+        run: dotnet restore ${{ env.SELENIUM_TEST_DIR }}
+
+      - name: Save cache for .NET packages
+        if: steps.has-cache.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v4
+        with:
+          key: ${{ env.DOTNET_CACHE_KEY }}
+          path: ${{ env.NUGET_CACHE_PATH }} 
+
+  build-and-test:
+    runs-on: im-linux
+    needs: nuget-cache
+    
+    env:
+      NUGET_CACHE_KEY: ${{ needs.nuget-cache.outputs.NUGET_CACHE_KEY }}
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.DISPATCH_REF }}
+
+      - name: Setup .NET # TODO: Delete if action runner contains the dotnet version you need.
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: ${{ env.DOTNET_VERSION }}
+
+      - name: Download NuGet Cache
+        id: restore-nuget-cache
+        uses: actions/cache/restore@v4
+        with:
+          key: ${{ env.NUGET_CACHE_KEY }}
+          enableCrossOsArchive: true
+          fail-on-cache-miss: true
+          path: ${{ env.NUGET_CACHE_PATH }}
+
+      - name: Build Project
+        shell: bash
+        run: dotnet build ${{ env.SELENIUM_TEST_DIR }} --configuration Release
+
+      - name: Run Selenium Tests
+        working-directory: '${{ env.SELENIUM_TEST_DIR }}'
+        shell: bash
+        env: 
+          # These environment variables can be accessed in C# using Environment.GetEnvironmentVariable("VARIABLE_NAME")
+          # and assigned in your config file to set configuration values dynamically.
+          ENVIRONMENT: ${{ inputs.environment }}
+          BROWSER: ${{ inputs.browser }}
+          APPLICATIONINSIGHTS_CONNECTION_STRING: ${{ secrets.APPLICATIONINSIGHTS_CONNECTION_STRING }}
+        run: |
+          dotnet test --configuration Release \
+          --logger "console;verbosity=detailed" \
+          --logger "trx;LogFileName=test-results.trx" \
+          --filter "${{ inputs.test-filter }}"
+
+      # # Uncomment the following step and remove the previous step to run tests specifically in the QualityPractices_Automation framework.
```

Comment:
I'm unfamiliar with what the `QualityPractices_Automation` is, and I might not be the only one. Adding a link where users can go read up on it might be helpful.

Reply:
Good suggestion! I’ve added a link to `QualityPractices_Automation `for quick reference.